# -----------------------------
# Options (maybe put as args)
# -----------------------------
alignmentOption: 0 # 0 - dense to dense, 1 - keypoints to dense, 2 - keypoints to keypoints
bShowAlignment: true
localisationOption: 2 # Option for localisation method (0 - PCL, 1 - RANSAC IA, 2 - Prerejective RANSAC)
keypointOption: 0

bPublishNurbsPointCloud: true # whether or not to publish the point cloud for visualuisation in Rviz

# -----------------------------
# Localisation
# -----------------------------
# Keypoints
/keypoints/modelResolution: 0.01
/keypoints/minNeighbours: 5

# Normals
pclNormalRadiusSetting: 0.05
# Features
pclFeatureRadiusSetting: 0.1
# RANSAC
/ransac/inlierMultiplier: 0.1
/ransac/maximumIterations: 5000

# POSSIBLY USE
/ransac/numberOfSamples: 3
/ransac/correspondenceRandomness: 3
/ransac/similarityThreshold: 0.9
/ransac/inlierFraction: 0.25

validInlierThreshold: 0.5


# Number of data points
nSurfPointsFactor: 5.0 # multiplier by the number of control points to get the number of surface points to sample


# -----------------------------
# Mapping
# -----------------------------
# Mesh processing
/meshing/numRowsDesired: 95
/meshing/numColsDesired: 95
/meshing/maxNanAllowed: 10
/meshing/removeNanBuffer: 2
/meshing/newRowColBuffer: 10

# NURBS generation
/mapping/useNonRectData: true # To use the new extension method
/mapping/nCtrlDefaultS: 17
/mapping/nCtrlDefaultR: 17

